website/docs/intro/migration/terraform-1.9.mdx

```diff
@@ -0,0 +1,178 @@
+---
+sidebar_position: 3
+sidebar_label: Migrating from Terraform 1.9.x
+description: |-
+    Learn how to migrate to OpenTofu from Terraform.
+---
+
+# Migrating to OpenTofu from Terraform 1.9.x
+
+When migrating from Terraform 1.9.x, please migrate to OpenTofu 1.9.0 first, then
+[upgrade your OpenTofu installation](/docs/intro/upgrading/) to the latest version.
+
+OpenTofu 1.9.0 is largely compatible with Terraform 1.9.x with a few minor changes needed. This migration guide will
+take you through the process of switching from Terraform to OpenTofu.
+
+## Step 0: Prepare a disaster recovery plan
+
+Although OpenTofu 1.9.0 is very similar to Terraform 1.9.8, make sure you have an up to date and *tested* disaster
+recovery plan.
+
+## Step 1: Upgrading Terraform
+
+This migration guide is only valid for Terraform 1.9.8. If you are on a Terraform version below 1.9.8, please upgrade
+to at least Terraform version 1.9.8 before proceeding with the migration by following the
+[Terraform upgrade guide](https://developer.hashicorp.com/terraform/language/v1.9.x/upgrade-guides). If you are on a
+higher Terraform version, please wait until an appropriate migration guide becomes available.
+
+## Step 2: Apply all changes with Terraform
+
+Before proceeding, make sure that you apply all changes with `terraform apply`. Running `terraform plan` should result
+in no planned changes. While you can switch to OpenTofu with pending changes, it is not recommended.
+
+```
+$ terraform plan
+
+...
+
+No changes. Your infrastructure matches the configuration.
+
+Terraform has compared your real infrastructure against your
+configuration and found no differences, so no changes are needed.
+```
+
+## Step 3: Install OpenTofu 1.9.0
+
+As a first step, please [follow the installation instructions for the OpenTofu CLI tool](intro/install/index.mdx). Please test
+if you can successfully execute the `tofu` command:
+
+```
+$ tofu --version
+OpenTofu v1.9.0
+on linux_amd64
+```
+
+## Step 4: Back up your state file and code
+
+Before you begin using the `tofu` binary on your Terraform code, make sure to back up your state file. If you are using
+a local state file, you can simply make a copy of your `terraform.tfstate` file in your project directory.
+
+If you are using a remote backend such as an S3 bucket, make sure that you follow the backup procedures for the
+backend and that you exercise the restore procedure at least once.
+
+Additionally, make sure you back up or version your code as the migration will require some code changes.
+
+## Step 5: Code changes
+
+For a successful migration, you will need to perform the following code changes
+
+### Functions
+
+The following functions are not supported in OpenTofu:
+
+- `encode_tfvars`
+- `decode_tfvars`
+- `encode_expr`
```

Comment:
We might include [these functions](https://github.com/opentofu/opentofu/issues/1614) in 1.9